plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/SiriusDiagramByRepresentationAndEObjectProvider.java

```diff
@@ -46,158 +56,195 @@
  * @author Romain Guider
  */
 public class SiriusDiagramByRepresentationAndEObjectProvider extends DiagramProvider {
-	/**
-	 * The option key used to retrieve the {@link EObject} root of some diagram representations. 
-	 */
-	private static final String KEY_TARGET_ROOT_OBJECT = "rootObject";
-	/**
-	 * The key used in the map passed to {@link IProvider} to define the Sirius
-	 * representation title from which image will be computed.
-	 */
-	private static final String KEY_DIAGRAM_DESCRIPTION_NAME = "diagramDescriptionName";
-
-	/**
-	 * Creates a new {@link SiriusDiagramByRepresentationAndEObjectProvider} given an EObject which
-	 * is the root of the seek diagram and the name of the diagram.
-	 * 
-	 * @param eObject
-	 *            the root of the diagram
-	 * @param title
-	 *            the name of the diagram
-	 * @param projectRootPath
-	 *            the project root path
-	 */
-	public SiriusDiagramByRepresentationAndEObjectProvider() {
-
-	}
-
-	/**
-	 * Replace forbidden characters with "_" in a filename
-	 * 
-	 * @param name
-	 * @return sanitized string
-	 */
-	public static String sanitizeFilename(String name) {
-		return name.replaceAll("[:\\\\/*?|<>]", "_");
-	}
-
-	/**
-	 * Get the full path (starting from the root folder) of the generated image
-	 * for a diagram
-	 * 
-	 * @param diagram
-	 * @param rootPath the project root path from which document generation has been launched.
-	 * @return Full path starting from root folder
-	 */
-	private String getDiagramImageFilename(DSemanticDiagram diagram, String rootPath) {
```

Comment:
I already mentionned that this code should be factored in a super class. I guess it was in the other part of the commit.